Table of Contents

    Introduction
    Prerequisites
    Tutorial
    Reference
    Getting help
    Frequently asked questions (FAQs)

I. Introduction
This is a bash-based pipeline for analysis of PacBio Iso-Seq generated with RSII or Sequel. Collectively, the pipeline will do:
A.	For RSII, extract raw reads, subreads and CCS reads from raw data in h5; For Sequel, only extract the CCS reads.
B.	Do some metadata analysis and give statistic reports, like adapter, raw reads length, raw reads quality, etc.
C.	Get full-length reads according to primer sequences and polyA. Completed sequences, which may represent the whole transcripts, are generated by cutting primer and polyA.
D.	Do some pre-mapping evaluation, like GC content of read, GC content across read, base quality across read.
E.	Mapping full-length reads onto genome.
F.	Filter and refine the raw alignments generated by mapping, like removing circular and translocated alignments, remove alignments with low coverage and identity, etc.
G.	Do some post-mapping evaluation, like read-length distribution, distance to TSS/TTS, etc.
H.	PA analysis, like distribution of cleavage site distance, motif around PA site, etc.
I.	Identify Alternative RNA Processing Events (APE, including SE, IR, A5SS, A3SS, APA) with CCS reads and junctions from RNA-seq (if available).
J.	APE characterization, like motif of splicing site, classifying APE into annotated or novel events, etc.
K.	Combination pattern of APE.

